src/BonjourMDNSManager.cpp

```diff
@@ -453,6 +454,7 @@ class MDNSManager::PImpl
                 std::string name = decodeDNSName(fromDnsSdStr(fullname));
                 std::string suffix = std::string(".") + rr->type + rr->domain;
                 std::string host = fromDnsSdStr(hosttarget);
+                hostent* host_struct = gethostbyname(hosttarget);
```

Comment:
Quoting the manual:
> The gethostbyname*(), gethostbyaddr*(), herror(), and hstrerror()
       functions are obsolete.  Applications should use [getaddrinfo(3)](https://man7.org/linux/man-pages/man3/getaddrinfo.3.html),
       [getnameinfo(3)](https://man7.org/linux/man-pages/man3/getnameinfo.3.html), and [gai_strerror(3)](https://man7.org/linux/man-pages/man3/gai_strerror.3.html) instead.